sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -124,6 +124,51 @@ struct snd_soc_ops max_98373_ops = {
 };
 EXPORT_SYMBOL_NS(max_98373_ops, SND_SOC_INTEL_SOF_MAXIM_COMMON);
 
+/*
+ * Look up present on board Maxim codecs hid and uid.
+ *
+ * uid: unique id of device. 0,1,2,3 total maximum 4 codecs.
+ * hid: hardware id of device. Current Maxim amps model support.
+ * amp_nums: return the codec numbers of first acpi device.
+ */
+static int get_maxim_num_codec(void)
+{
+	const char * const UID_STRINGS[] = { "0", "1", "2", "3" };
+	const char * const MAXIM_HID[] = {"MX98373", "MX98390"};
+
+	unsigned int uid, amp_nums = 0;
+	unsigned int hid, cur = 0;
+	struct acpi_device *adev;
+
+	/* iterate each uid if presents. */
+	for (uid = 0; uid < ARRAY_SIZE(UID_STRINGS); uid++) {
+		/* finding the first match codec HID */
+		for (hid = cur; hid < ARRAY_SIZE(MAXIM_HID); hid++) {
+			adev = acpi_dev_get_first_match_dev(MAXIM_HID[hid], UID_STRINGS[uid], -1);
```

Comment:
@plbossart. the purpose is getting the number of codecs at all.
Because max_98373/98390_dai_link(), sof_max98373/98390_codec_conf() did. So I didn't put into device here.